corehq/apps/hqwebapp/models.py

```diff
@@ -24,7 +24,7 @@ def __new__(cls, category, action, label=None):
         return super(GaTracker, cls).__new__(cls, category, action, label)
 
 
-class MaintenanceAlert(models.Model):
+class CommCareHQAlert(models.Model):
```

Comment:
Seems like most classes in HQ could have a `CommCareHQ` prefix, and that would be redundant. What do you think of renaming this to `Alert`?

Aside: IMO the model rename doesn't seem particularly important, and it introduced a lot of noise in the change set.

Reply:
Agreed I could lose the 'CommCareHQ" in the name.
I think this could be better called "Notice" or "MessageBanner". I feel alert has a sense of "alarm" that these notices would not always be. Any suggestions?
We already have notification model.

> the model rename doesn't seem particularly important,

Okay. I do think the rename was important/relevant since for domains it has nothing to do with maintenance. I didn't consider the issues it would create and went for it. Renaming a model is definitely not straight forward.

Reply:
> Did you decide not to rename this?

Not really. Was waiting for some follow up feedback on this.

> I think this could be better called "Notice" or "MessageBanner". I feel alert has a sense of "alarm" that these notices would not always be. Any suggestions?

I am unsure about using "Alert' because of the reason above. I think this feature is closer to a "banner" than an "alert", for instance, the project requesting this feature wants to show the user a data privacy message to the user each time they login. If that does not seem logical, I am happy to rename it to `Alert` only.

> Seems like most classes in HQ could have a CommCareHQ prefix, and that would be redundant.

Right. It does look like we have though, built a series of such things. There are numerous models with `CommCare` prefix.

Reply:
> Was waiting for some follow up feedback on this.

I don't have strong opinions on the name as long as it does not have the redundant prefix. `Alert` seems fine to me since it does create a pop up that is intended to alert the user to something. It seems better for the name to imply a bit of alarm rather than to downplay the fact that this might be annoying to the user (who likes popups).

> It does look like we have though, built a series of such things. There are numerous models with `CommCare` prefix.

I do not think this is a good pattern to follow. It violates the DRY principle.

Reply:
Cool,

Renamed it to Alert
https://github.com/dimagi/commcare-hq/pull/33592/commits/c2856a76caa79035d474df6fdfd54161fb704e31

One follow up,
I am thinking I should merge this new migration with the old one by deleting the new one and just modify the old one.
Though the old migration has been run on staging, so my plan is to let the new migration run on staging once and then do the merge/remove business.
I can't rollback the migration on staging because there are already some domain alerts added by QA for testing which would get messed up.
Just checking if that sounds like the right approach.

Reply:
Yes, please squash all migrations in this PR into a single migration.

The plan for staging sounds reasonable as long as the resulting database schema does not change as a result of squashing, which it should not. You may need to fake the new squashed migration on staging since it will have already been deployed. Also note that the application of the new squashed migration will not be tested on staging if you do it this way.